pallets/liquidity-pools-gateway/src/lib.rs

```diff
@@ -462,6 +619,140 @@ pub mod pallet {
 
 			Ok((address, incoming_msg))
 		}
+
+		/// Iterates over the outbound messages stored in the queue and attempts
+		/// to process them.
+		///
+		/// If a message fails to process it is moved to the
+		/// `FailedOutboundMessages` storage so that it can be executed again
+		/// via the `process_failed_outbound_message` extrinsic.
+		fn service_outbound_message_queue(max_weight: Weight) -> Weight {
+			let mut weight_used = Weight::zero();
+
+			let mut processed_entries = Vec::new();
+
+			for (nonce, (domain, sender, message)) in OutboundMessageQueue::<T>::iter() {
+				processed_entries.push(nonce);
+
+				let weight =
+					match Self::process_message(domain.clone(), sender.clone(), message.clone()) {
+						Ok(post_info) => {
+							Self::deposit_event(Event::OutboundMessageExecutionSuccess {
+								nonce,
+								sender,
+								domain,
+								message,
+							});
+
+							post_info
+								.actual_weight
+								.expect("Message processing success already ensured")
+								// Extra weight breakdown:
+								//
+								// 1 read for the outbound message
+								// 1 write for the event
```

Comment:
Dispatching an event is much cheaper than performing a `write` operation. But we can certainly keep the `1 write` here to be defensive.